### ISSUE TITLE
Issue when decrypting content

### ISSUE THREAD

**Issue by nicholascarballo** [axiomatic-systems/Bento4#523]
I am currently trying to decrypt an only video file using mp4decrypt. I already have the decryption key but when I decrypt the file the result says its a corrupted file when I try to play it on VLC.
I ran mp4info to check file before and after decryption and here is the results.

Before decryption
```
root@me:/home/myvideo# /home/Bento4/cmakebuild/./mp4info output.mp4
File:
  major brand:      dash
  minor version:    0
  compatible brand: iso2
  compatible brand: dash
  fast start:       yes

Movie:
  duration:   -1161359157 ms
  time scale: 10000000
  fragments:  yes

Found 1 Tracks
Track 1:
  flags:        7 ENABLED IN-MOVIE IN-PREVIEW
  id:           1
  type:         Video
  duration: -1161359157 ms
  language: und
  media:
    sample count: 0
    timescale:    10000000
    duration:     -1 (media timescale units)
    duration:     -1161359157 (ms)
    bitrate (computed): 273.581 Kbps
    sample count with fragments: 180
    duration with fragments:     60000000
    duration with fragments:     6000 (ms)
  display width:  398.000000
  display height: 224.000000
  Sample Description 0
    [ENCRYPTED]
      Coding:         encv
      Scheme Type:    cenc
      Scheme Version: 65536
      Scheme URI:
    Coding:      avc3 (H.264)
    Width:       398
    Height:      224
    Depth:       24
    AVC Profile:          100 (High)
    AVC Profile Compat:   0
    AVC Level:            13
    AVC NALU Length Size: 4
    AVC SPS: [6764000dac3ca506477afff070006ff420202028000003000800000301e700000afc80015f91f89f05]
    AVC PPS: [68ef7525]
    Codecs String: avc3.64000D
```

After decryption
```
root@me:/home/myvideo# /home/Bento4/cmakebuild/./mp4info decrypted.mp4
File:
  major brand:      dash
  minor version:    0
  compatible brand: iso2
  compatible brand: dash
  fast start:       yes

Movie:
  duration:   -1161359157 ms
  time scale: 10000000
  fragments:  yes

Found 1 Tracks
Track 1:
  flags:        7 ENABLED IN-MOVIE IN-PREVIEW
  id:           1
  type:         Video
  duration: -1161359157 ms
  language: und
  media:
    sample count: 0
    timescale:    10000000
    duration:     -1 (media timescale units)
    duration:     -1161359157 (ms)
    bitrate (computed): 273.581 Kbps
    sample count with fragments: 180
    duration with fragments:     60000000
    duration with fragments:     6000 (ms)
  display width:  398.000000
  display height: 224.000000
  Sample Description 0
    Coding:      avc3 (H.264)
    Width:       398
    Height:      224
    Depth:       24
    AVC Profile:          100 (High)
    AVC Profile Compat:   0
    AVC Level:            13
    AVC NALU Length Size: 4
    AVC SPS: [6764000dac3ca506477afff070006ff420202028000003000800000301e700000afc80015f91f89f05]
    AVC PPS: [68ef7525]
    Codecs String: avc3.64000D
```

My output.mp4 is generated using cat.

`cat init.mp4 segment1.mp4 > output.mp4`

to decrypt this file I use

`mp4decrypt —key mykey output.mp4 decrypted.mp4`

What am I possibly doing wrong? My bento4 tools files are up to date, downloaded them 3 days ago from this same repo.

Thanks in advance.

**Comment by barbibulle** [comment]
The command-line syntax for the key argument of `mp4decrypt` is `--key <id>:<key>`, where `<id>` is the track ID, so `1` here.
There's no way to check whether the key you are using is correct or not, other than checking that the decrypted output is valid or invalid (there's no "checksum"-like property to check the value of the key). Do you know for sure your decryption key is correct? Did you encrypt the file yourself? If so, what command-line did you use for that?

**Comment by nicholascarballo** [comment]
I didn't encrypt the file myself but I have the right key to decrypt. And I tried --key 1:key parameter and also it didn't worked.
mp4info shows the file is decrypted, but why it doesn't play? I tried different files and all keys I have are from the same license provider. Some work, some don't.

Heres is what I see on VLC 
https://i.imgur.com/7ZG1A2h.png

I wonder if this is something related to mp4decrypt itself.

**Comment by barbibulle** [comment]
There's an edge case that I fixed earlier this week, maybe that's what you've run into. Could you try with the latest release to check if you're still having the same issue?

**Comment by nicholascarballo** [comment]
I've deleted all the files and rebuild them from scratch. Tried and the results stays the same. I see on VLC gray, white and sometimes black screen and with different sizes.

**Comment by barbibulle** [comment]
Would it be possible to share a link to one of the files along with the decryption key, so that I can try to replicate the problem on my side?

**Comment by nicholascarballo** [comment]
Yes, of course.
Theres init for audio and video along with 7 segments for each stream. Decryption in format kid:key is inside .txt file.

https://mega.nz/file/MJUkVKbC#pfiDeVcg3wTl9guK896w9SsK-owuG5eWUImZUCzE-vU

**Comment by barbibulle** [comment]
The output, when decrypting, is clearly completely mangled. This means that either the key isn't the right one, or somehow the tool that encrypted the frames and `mp4decrypt` have a different understanding of the spec. Without an example with both the unencrypted video and the encrypted video, it's impossible to tell.
Maybe there's some other way to compare: do you have a player that is capable of playing back this encrypted file?

**Comment by nicholascarballo** [comment]
Yes. Can I send a link to a webplayer that plays the content in your e-mail? I don't want the link here.

**Comment by barbibulle** [comment]
Yes, that would be fine. You can also use the Zendesk support page on http://www.bento4.com

**Comment by nicholascarballo** [comment]
I've just sent you the webpage through Zendesk support page. The subject is "Issue when decrypting content #523"

**Comment by nicholascarballo** [comment]
After decrypting the content and merging audio and video I get this log on ffmpeg.

[mov,mp4,m4a,3gp,3g2,mj2 @ 0x55555602dfc0] st: 0 edit list: 1 Missing key frame while searching for timestamp: 666664
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x55555602dfc0] st: 0 edit list 1 Cannot find an index entry before timestamp: 666664.
[h264 @ 0x5555560927c0] SEI type 80 size 1440 truncated at 246
[h264 @ 0x5555560927c0] A non-intra slice in an IDR NAL unit.
[h264 @ 0x5555560927c0] decode_slice_header error
[h264 @ 0x5555560927c0] no frame!
[h264 @ 0x5555560927c0] SEI type 176 size 984 truncated at 127
[h264 @ 0x5555560927c0] non-existing PPS 2 referenced
[h264 @ 0x5555560927c0] decode_slice_header error
[h264 @ 0x5555560927c0] no frame!
[aac @ 0x55555608d000] Reserved bit set.
[aac @ 0x55555608d000] Prediction is not allowed in AAC-LC.
[h264 @ 0x5555560927c0] SEI type 163 size 1120 truncated at 127
[h264 @ 0x5555560927c0] slice type 12 too large at 6
[h264 @ 0x5555560927c0] decode_slice_header error
[h264 @ 0x5555560927c0] no frame!
[h264 @ 0x5555560927c0] SEI type 217 size 984 truncated at 124
[h264 @ 0x5555560927c0] non-existing PPS 2 referenced
[h264 @ 0x5555560927c0] decode_slice_header error
[h264 @ 0x5555560927c0] no frame!
[h264 @ 0x5555560927c0] chroma_format_idc 5 is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.
[h264 @ 0x5555560927c0] If you want to help, upload a sample of this file to ftp://upload.ffmpeg.org/incoming/ and contact the ffmpeg-devel mailing list. (ffmpeg-devel@ffmpeg.org)
[h264 @ 0x5555560927c0] chroma_format_idc 5 is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.
[h264 @ 0x5555560927c0] If you want to help, upload a sample of this file to ftp://upload.ffmpeg.org/incoming/ and contact the ffmpeg-devel mailing list. (ffmpeg-devel@ffmpeg.org)
[h264 @ 0x5555560927c0] chroma_format_idc 5 is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.
[h264 @ 0x5555560927c0] If you want to help, upload a sample of this file to ftp://upload.ffmpeg.org/incoming/ and contact the ffmpeg-devel mailing list. (ffmpeg-devel@ffmpeg.org)
[h264 @ 0x5555560927c0] SEI type 38 size 688 truncated at 246
[h264 @ 0x5555560927c0] concealing 4 DC, 4 AC, 4 MV errors in I frame
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x55555602dfc0] Auto-inserting h264_mp4toannexb bitstream filter
frame=  318 fps= 30 q=-1.0 size=N/A time=00:00:10.62 bitrate=N/A speed=   1x


I read somewhere in stackoverflow that my files could be possibly not h264 compliant. Is this possible? If so how do I check if the files are correct or how do I fix them?
